python/ngen_cal/src/ngen/cal/model.py

```diff
@@ -213,6 +217,23 @@ class ModelExec(BaseModel, Configurable):
     args: Optional[str]
     workdir: DirectoryPath = Path("./") #FIXME test the various workdirs
     eval_params: Optional[EvaluationOptions] = Field(default_factory=EvaluationOptions)
+    plugins: List[PyObjectOrModule] = Field(default_factory=list)
+    plugin_settings: Dict[str, Dict[str, Any]] = Field(default_factory=dict)
+
+    _plugin_manager: PluginManager
+    class Config(BaseModel.Config):
+        # properly serialize plugins
+        json_encoders = {
+            type: type_as_import_string,
+            ModuleType: lambda mod: mod.__name__,
+            FunctionType: type_as_import_string,
+        }
+        underscore_attrs_are_private = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(**kwargs)
+        model_plugins = cast(List[Union[Callable, ModuleType]], self.plugins)
+        self._plugin_manager = setup_plugin_manager(model_plugins)
```

Comment:
I get the desire to setup the plugin manager here, but I would rather not. Why not just keep all the setup code in a few small easy to follow functions near `main`?

Reply:
Unrelated to my first comment, I am getting the feeling that we should just create a top level `plugins` and `plugin_settings` keys. So remove it from `general` and `model` and just move it to the top level. I think it will be the clearest and simplest to just have them all at the top level. Thoughts?

Reply:
will leave this for now, and keep in mind plugin management, scope, ect as we continue to build on the concept.